src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,219 @@
+     */
+    protected void markItemDone(String input) {
+        try {
+            String[] words = input.split(" ");
+            if (words.length > 2) {
+                throw new DukeException("too many parameters!");
+            }
+            int index = Integer.parseInt(words[1]);
+            tasks.get(index - 1).markDone();
+            System.out.println("cool, this task is marked as done");
+            System.out.println("\t" + tasks.get(index - 1));
+        } catch (NumberFormatException | ArrayIndexOutOfBoundsException | DukeException e) {
+            System.out.println("format: mark <number>");
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("enter a valid index");
+        }
+    }
+
+    /**
+     * Parses String in the form of mark x, marking index x in the list as not done.
+     *
+     * @param input String that contains information about item to be marked not done.
+     */
+    protected void markItemUndone(String input) {
+        try {
+            String[] words = input.split(" ");
+            if (words.length > 2) {
+                throw new DukeException("too many parameters!");
+            }
+            int index = Integer.parseInt(words[1]);
+            tasks.get(index - 1).markUndone();
+            System.out.println("ok, this task is marked as not done yet");
+            System.out.println("\t" + tasks.get(index - 1));
+        } catch (NumberFormatException | ArrayIndexOutOfBoundsException | DukeException e) {
+            System.out.println("format: mark <number>");
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("enter a valid index");
+        }
+    }
+
+    /**
+     * Parses String in the form of mark x, deleting task at index x in the list.
+     *
+     * @param input String that contains information about item to be deleted.
+     */
+    protected void deleteTask(String input) {
+        try {
+            String[] words = input.split(" ");
+            if (words.length > 2) {
+                throw new DukeException("too many parameters!");
+            }
+            int index = Integer.parseInt(words[1]);
+            tasks.remove(index - 1);
+            System.out.println("ok, i removed this task");
+        } catch (NumberFormatException | ArrayIndexOutOfBoundsException | DukeException e) {
+            System.out.println("format: mark <number>");
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("enter a valid index");
+        }
+    }
+
+    protected void find(String s) {
+        try {
+            String toFind = s.substring(5);
+            ArrayList<Task> matchingTasks = new ArrayList<>();
+            for (int i = 0; i < tasks.size(); i++) {
+                Task task = tasks.get(i);
+                if (task.description.contains(toFind)) {
+                    matchingTasks.add(task);
+                }
+            }
```

Comment:
Maybe you can consider moving this part to a new method in TaskList class that returns an instance of TaskList with matching tasks?